### DISCUSSION TITLE
Text editing

### DISCUSSION THREAD

**Discussion by donno2048** [GitSquared/edex-ui#973] [Ideas]
Edit a text file by simply clicking on it would be great...

It can open a .txt so why can't it open a .py or .rb etc?

**Comment by Animus-Surge** [comment]
I’ll add those file types to the editor list.

*admin edit: see #975*

**Reply by donno2048** [reply]
Why can't it recognize text files?

**Reply by GitSquared** [reply]
Indeed. I've recommended that we use that approach instead of managing every file extension. You can track the work on that here: #975

**Reply by branislavblazek** [reply]
can I open editor for .txt file by some command? I can open editor by clicking on file in filessytem explorer.

**Reply by Animus-Surge** [reply]
> can I open editor for .txt file by some command?

The idea for my text editor is to use it with the filesystem view. It's very hard to add commands into the terminal (especially for windows) that works with eDEX, so unfortunately it is not possible to do the command.

>I am on Windows. I cannt open any terminal-editor like nano or vim. When I click in filesystem on some .txt file, Edex UI opens a window with content of that file. How i can achieve opening that window by some command or hot key? I didn't found yet any command or hotkey how to open that window.

(comment link: https://github.com/GitSquared/edex-ui/discussions/973#discussioncomment-367342)

Expanding on this comment, I could probably add a hotkey that opens a blank text editor, and I could add another button onto the editor that opens a dialogue asking for the filepath, either absolute or relative to the current filesystem directory. Alternatively, there are windows versions of vim that you can download, or you can install windows subsystem for linux (wsl) to be able to use nano.

**Reply by ghost** [reply]
It would be straightforward to add a handler for ANSI escape sequences via `terminal.parser.registerOscHandler(1024, handler)` in `terminal.class.js`, where 1024 is some unused OSC handler code and `handler: string => bool` returns whether the handler is done with the data (`true`) or wants to give some other handler a go at it (`false`).  Then you can do something like `printf "\e]1024;<string data here>\a"` to pass data to the handler.  I recommend using JSON for the data.

**Reply by ghost** [reply]
Here's a proof of concept: https://github.com/GitSquared/edex-ui/pull/1163

**Comment by kamalov69** [comment]


**Reply by kamalov69** [reply]
What you say?

**Comment by inthewallstonight** [comment]
Depends on your text editor

**Reply by branislavblazek** [reply]
I am on Windows. I cannt open any terminal-editor like nano or vim. When I click in filesystem on some .txt file, Edex UI opens a window with content of that file. How i can achieve opening that window by some command or hot key? I didn't found yet any command or hotkey how to open that window.
![image](https://user-images.githubusercontent.com/59296705/107873354-4b360e80-6eb2-11eb-99d2-78a10e819874.png)
On this image you see the Edex UI's editor. I want open it with command or hot key.

**Reply by GitSquared** [reply]
I think ctrl+click (or shift+click, not sure) on the file icon will open it using your OS' default program.

**Reply by JacobLiu001** [reply]
I am on Windows 10 and have full control over my device. 
Ctrl+Clicking on any file would result in an error message. It might be cause by a change in electron APIs? 
I have found this link at Electron's governance docs: https://github.com/electron/governance/blob/master/wg-api/spec-documents/shell-openitem.md
Maybe a small fix? I'm a n00b coder and don't know any electron... 

![image](https://user-images.githubusercontent.com/28747843/110326230-544e7180-8053-11eb-86da-87e376da179e.png)
 In this image I ctrl+clicked a file in the file explorer in the lower half of the screen.

**Reply by GitSquared** [reply]
Indeed, looks like I missed the breaking change. Thanks for linking to the electron docs, really helpful. I'm going to fix this in master rn and we *should* have a patch release this week.

(edit: ref 417a82e6c109d408c11155997fc82d46537f769f )

**Reply by ghost** [reply]
Fixed in https://github.com/GitSquared/edex-ui/pull/1159
